### PR TITLE
fix: avoid static node restarts from sudo warnings

### PR DESCRIPTION
**Pull request by Levi080513**
## Issue

NEU-515

TestRail: C2728950

## Summary

Static node config hash checks can receive stderr text together with stdout because SSH command execution uses combined output. On nodes where sudo prints hostname warnings, unchanged config files were misdetected as changed, causing restart-sensitive components such as accelerator-exporter and vmagent to be recreated every reconcile. This PR parses only valid SHA-256 tokens from successful remote hash output and logs the exact reason whenever a static node component is restarted.

## Changes

- Parse remote SHA-256 output defensively in `pkg/command_runner.FileClient`, ignoring sudo/SSH warning text while preserving command failure behavior.
- Cover edge cases for warning-before/after-hash output, uppercase hashes, hash-looking path fields, empty output, and invalid/truncated hashes.
- Add static-node restart logging before `RestartComponentContainer`, including component identity, desired hash, image, `config_changed`, `container_matches`, and reason.
- Add tests for restart reason composition and remote SHA-256 parsing edge cases.

### Unit test

- [x] `go test ./pkg/command_runner ./internal/cluster/staticnode`
- [x] `git diff --check`
- [x] GitHub `pr-check` passed on head `90ce00136ba9956bffe87f5c0ce545e7f6432dc4`

### DB test

- [x] Not affected. No DB schema, migration, storage query, or persisted contract changes.

### E2E test

- [x] Not required. The root cause is deterministic remote hash output parsing; sudo warning and Docker restart effects are covered by unit tests around `WriteFileIfChanged` and static-node restart reason construction. Manual E2E is intentionally not required because it depends on host-specific sudo/hostname warning state and Docker event timing.

## Risk & Rollback

Risk is limited to static-node remote config comparison and restart observability. Invalid or missing remote hash output remains conservative and writes the file. Rollback by reverting this PR; no DB or public API rollback required.

### CODE REVIEW

File: internal/cluster/staticnode/reconciler.go [hunk 1]
```diff
@@ -556,6 +571,48 @@ func (r *Reconciler) reconcileComponent(
...
+		reasons = append(reasons, "config changed")
+	}
+
+	if !containerMatches {
+		reasons = append(reasons, "container does not match desired state")
+	}
+
+	if inspectErr != nil {
+		reasons = append(reasons, "container inspect failed: "+inspectErr.Error())
+	}
```
[the remark this answers is not in the document]
Comment (Levi080513) [reply internal/cluster/staticnode/reconciler.go]: Fixed in `cc6b34d7`: `componentRestartReason` now appends `container does not match desired state` only when `inspectErr == nil`. Inspect failures now log `container inspect failed: ...` without inferring mismatch, while `config changed` is still preserved when both conditions apply. Added the `config changed and inspect error` test case and updated the design doc accordingly.

**Review by Yuyz0112 — APPROVED** [review]
